### PR TITLE
Reviewing

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: project/REVIEWING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+This is to avoid the pain of making a reviewer rework their whole PR and then not merging it.
+
+Things to hunt for :
+
+- missing docs . This is what you should look for first. If you think the PR lacks docs,
+ask for them, because you will be better at reviewing it if you understand it better,
+and docs help a lot with that.
+- missing tests : Encourage people to do TDD by making clear a PR will not get merged
+if it lacks tests. Not everything is easy to test though, keep that in mind.
+- BC breaks : If there is a BC-break, ask for a deprecation system to be created instead.
```
Comment (OskarStark) [inline project/REVIEWING.md]: ... and check if the correct branch is used ?

File: project/REVIEWING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+- Unclear pieces of code : does the code use clear, appropriate variable or class names,
+or does it use names like `data`,`result`, `WhateverManager`, `SomethingService`?
```
Comment (OskarStark) [inline project/REVIEWING.md]: missing space after `data,`

File: project/REVIEWING.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Are exception names still meaningful if you remove the `Exception` suffix?
```
Comment (OskarStark) [inline project/REVIEWING.md]: should have a message too

File: project/REVIEWING.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+Is the contributor trying to be clever or to be clear?
+- Violations of the SOLID principles :
```
Comment (OskarStark) [inline project/REVIEWING.md]: should be a link to: https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)

**Comment by OskarStark** [comment]
check the dependencies, if something new is introduced?

File: project/REVIEWING.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+Leave no stone unturned. When in doubt, ask for a clarification. If the
+clarification seems useful, and does not appear in a code comment or in a commit
+message, say so and / or make use a squash-merge to customize the commit message.
+Ideally, the project history should be understandable without an internet connection,
+and the PR should be understandable without having a look at the changes.
+
+Also, make sure your feedback is actionable, it is important to keep the ball rolling,
+so if you raise a question, try to also provide solutions.
+
+## Labelling the PR
```
Comment (OskarStark) [inline project/REVIEWING.md]: this can only be done by core members if i am right, but the can ask us to apply labels

File: project/REVIEWING.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+or does it use names like `data`,`result`, `WhateverManager`, `SomethingService`?
+Are exception names still meaningful if you remove the `Exception` suffix?
+Is the contributor trying to be clever or to be clear?
+- Violations of the SOLID principles :
+    - S : If a class is 3000 lines long, maybe it does too many things?
+    - O : Is there a big switch statement that might grow in the future?
+    - L : Does the program behave reasonably when replacing a class with a child class?
+    - I : Are interfaces small and easy to implement? If not, can they be split into smaller interfaces?
+    - D : Is the name of a class hardcoded in another class, with the `new` keyword or a static call?
+- Spelling / grammar mistakes, including in commit messages or said notes.
```
Comment (OskarStark) [inline project/REVIEWING.md]: never heard `said notes`, did you mean `side notes` or is it correct?

File: project/REVIEWING.md (hunk 1 above)
Comment (greg0ire) [inline project/REVIEWING.md]: This bears repeating indeed

File: project/REVIEWING.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+so if you raise a question, try to also provide solutions.
+
+## Labelling the PR
+
+There are several labels that will help determine what the next version number will be.
+Apply the first label that matches one of this conditions, in that order:
+
+- `major`: there is a BC-break. The PR should target the `master` branch.
+- `minor`: there is a change in the API. The PR should target the stable branch.
+- `patch`: this fixes an issue (not necessarily reported).  The PR should target the stable branch.
```
Comment (OskarStark) [inline project/REVIEWING.md]: 2 spaces after `reported).`

**Comment by OskarStark** [comment]
Nice work @greg0ire 👍

File: project/REVIEWING.md (hunk 6 above)
Comment (greg0ire) [inline project/REVIEWING.md]: It's correct, it is meant to refer to something that was said previously. In this occurrence, I talked about notes in a previous sentence. And then I removed it :confounded: . So I guess I'll simplify this!

**Comment by greg0ire** [comment]
Thanks for the review @OskarStark , I think I fixed everything you said!

**Comment by OskarStark** [comment]
> Thanks for the review @OskarStark , I think I fixed everything you said!

"Ich könnte dich knuddeln" 😄

File: project/REVIEWING.md (hunk 6 above)
Comment (OskarStark) [inline project/REVIEWING.md]: > it is meant to refer to something that was said previously

makes sense

**Comment by OskarStark** [comment]
btw Contributing.md and REVIEWING.md should be on the website too IMO, because this is something about evolving the project with quality

**Comment by Soullivaneuh** [comment]
Looks great, have to read it carefully. But why put it on a separate file? Reviewing is part of the contributing process IMHO. :wink:

**Comment by greg0ire** [comment]
> Looks great, have to read it carefully. But why put it on a separate file? Reviewing is part of the contributing process IMHO. :wink:

Because the CONTRIBUTING file is already very very long and I'd like not to bug "simple" contributors with things they do not care about.

**Comment by Soullivaneuh** [comment]
> Because the CONTRIBUTING file is already very very long and I'd like not to bug "simple" contributors with things they do not care about.

This is also why summaries exist. :trollface:

**Comment by greg0ire** [comment]
What are you suggesting exactly here? Should I write a summary in CONTRIBUTING.md, linking to REVIEWING.md?

**Comment by Soullivaneuh** [comment]
@greg0ire No. Have a Reviewing section under the Contributing folder.

And then add a link here: https://github.com/sonata-project/dev-kit/blob/master/project/CONTRIBUTING.md#summary

File: project/CONTRIBUTING.md [hunk 8]
```diff
@@ -299,3 +302,86 @@ This is a consensus made on #4242 in addition to #1337.
...
+
+If there are several commits in a PR, make sure you review it commit by commit,
+so that you can check the commit messages, and make sure the commit are independent
+and atomic.
+
+### Merging
+
+Do not merge something you wrote yourself. Do not merge a PR you reviewed alone,
+instead, merge PRs that have already be reviewed and approved by another reviewer.
+You can give the green light to a PR by commenting "LGTM" ("Looks Good To Me") on it.
```
Comment (core23) [inline project/CONTRIBUTING.md]: We don't need the LGTM comment anymore

Comment (greg0ire) [inline project/CONTRIBUTING.md]: Yeah, that's how old this PR is… :spider_web:

Comment (greg0ire) [inline project/CONTRIBUTING.md]: Fixed

**Comment by core23** [comment]
Waiting for decision about the right place in
https://github.com/sonata-project/dev-kit/pull/227

**Comment by jordisala1991** [comment]
I am 👍  for this doc, most of this should apply to every project not only Sonata. Should we add a section for BC? I don't know if it should be on CONTRIBUTING or REVIEWING, or maybe on another that gets linked from both.

IMO BC is an important part of the review.

**Comment by greg0ire** [comment]
I think we already have something in CONTRIBUTING.md

File: project/CONTRIBUTING.md [hunk 9]
```diff
@@ -2,11 +2,14 @@
 
 Thanks for your interest in Sonata projects!
 
+This document is about issues and pull requests.
+If you want to contribute a code review see [the dedicated document](./REVIEWING.md).
```
Comment (Soullivaneuh) [inline project/CONTRIBUTING.md]: I think this part is not relevant anymore.

**Comment by Soullivaneuh** [comment]
I'll trust @greg0ire for this one but I would like some other approvals so I can merge this one.

Especially @core23 who blocked this PR.

File: project/CONTRIBUTING.md [hunk 10]
```diff
@@ -338,6 +340,88 @@ We agreed that blank color is boring and so deja vu. Pink is the new way to do.
...
+or if there is a situation (faulty commit, revert needed) maybe you can merge it directly.
+
+### Be nice to the contributor
+
+Thank them for contributing. Encourage them if you feel this is going to be long.
+In short, try to make them want to contribute again. If they are stuck, try to provide them with
+code yourself, or ping someone who can help.
+
 [sphinx_install]: http://www.sphinx-doc.org/en/stable/
 [pip_install]: https://pip.pypa.io/en/stable/installing/
```
Comment (jordisala1991) [inline project/CONTRIBUTING.md]: Does it makes sense to have this here? Seems like the continuation of `###Be nice to the contributor` part

**Comment by greg0ire** [comment]
@sonata-project/contributors can you please review again?

File: project/CONTRIBUTING.md (hunk 10 above)
Comment (greg0ire) [inline project/CONTRIBUTING.md]: @Soullivaneuh requested to have all links at the end of the document, in the PR that introduced these links.

Comment (Soullivaneuh) [inline project/CONTRIBUTING.md]: @jordisala1991 Links may be used on other parts of the document.

**Comment by Soullivaneuh** [comment]
I still need @core23 approval. :wink:

File: project/CONTRIBUTING.md [hunk 11]
```diff
@@ -338,6 +340,88 @@ We agreed that blank color is boring and so deja vu. Pink is the new way to do.
...
+so if you raise a question, try to also provide solutions.
+
+### Labelling the PR
+
+Applying labels requires write access to PRs, but you can still advise if you do not have them.
+There are several labels that will help determine what the next version number will be.
+Apply the first label that matches one of this conditions, in that order:
+
+- `major`: there is a BC-break. The PR should target the `master` branch.
+- `minor`: there is a change in the API. The PR should target the stable branch.
```
Comment (core23) [inline project/CONTRIBUTING.md]: Maybe mention that this would apply to BC features?

**Comment by core23** [comment]
Reviewed! @Soullivaneuh

**Pull request merged**